### ISSUE TITLE
Could not perform action reset for Faceted Search module

### ISSUE THREAD

**Issue by banan44** [PrestaShop/PrestaShop#15966]
Hello,
I am facing problems with Faceted Search recently. I cannot update indexes. When I start updating I get 'indexing failed' error after a while. I wanted to Reset the module and configure it again but with no luck.

![obraz](https://user-images.githubusercontent.com/48493952/66846980-ef344080-ef72-11e9-9bea-28624201504d.png)

I looked at logs and all errors I have are "content-length(0): ReceiveAckHdr: timeout 300 is exceeded" fo faceted search module. Is there any way to make it work?

**Comment by khouloudbelguith** [comment]
Hi @banan44,

Could you please try to re-install the module by removing the old module during the uninstall & try to configure it by editing the template & rebuilding the entire price index & building attributes & features indexes & clearing the cache.
![image](https://user-images.githubusercontent.com/16067358/66762400-37396180-eea6-11e9-8826-5948e5e78713.png)

Thanks!

**Comment by banan44** [comment]
Hi @khouloudbelguith, I've just tried that. Uninstalling the module was an easy part with success. When I want to install the module from Module Catalog I have this error

![obraz](https://user-images.githubusercontent.com/48493952/66863023-72b15a00-ef92-11e9-84b7-8a9dadcd1e4e.png)

The module appears in Module Manager though. I've created a new template and Indexed all missing prices first. No luck. I had 'Indexing is in progress. Please do not leave this page' for a few minutes:

![obraz](https://user-images.githubusercontent.com/48493952/66863603-ae005880-ef93-11e9-9008-5c7d60cd06be.png)

Then I have following error:

![obraz](https://user-images.githubusercontent.com/48493952/66863639-bc4e7480-ef93-11e9-80ff-8f91f16a2696.png)

The errors from logs are "timeout 300 is exceeded" again. These errors appeared during installation and rebuilding.

FYI. I have multistore. The problem started when I upgraded PS to 1.7.6.0. I could not rebuild indexes since that moment. I though this is a temporary issue but 2 day ago I've updated PS to 1.7.6.1 and the problem persists.

**Comment by khouloudbelguith** [comment]
@banan44, I did not manage to reproduce the issue with PS1.7.6.1 & Multistore context is enabled.
https://drive.google.com/file/d/1R0qxv6HN5S2gFzwiGpJ9G_BLqJKnGKpr/view
What is the exact version of the module ps_facetedsearch did you use?

Thanks!

**Comment by banan44** [comment]
@khouloudbelguith the latest one. I wrote in a previous message that I installed it from Module Catalog after uninstalling it the version I had (which also was the latest one).

**Comment by khouloudbelguith** [comment]
@banan44,

Could you please provide us with more info? We need more details to understand how we can reproduce your issue:
 - server setup and configuration
 - PrestaShop version (source)
 - debug mode report
 - PHP error logs
 - apache error log
 - javascript console log

Don't you know how to get this information? Please read the following article:
http://build.prestashop.com/howtos/misc/how-to-create-bug-report/

Thanks!

**Comment by banan44** [comment]
@khouloudbelguith are these information good for you? 
Link: https://drive.google.com/open?id=1dkLxUM1WN14ahDGJY2YaW0AhmIA5F3_A

I also included a screenshot to show you how many Features I have. I included logs from main domain and from the second shop. Todays logs are empty because I have not updated indexes yet. I've updated the Faceted Search indexes while being phisically logged on second shop (second URL). It's because the second URL does not have Cloudflare yet that could interrupt the update procedure. But I also tried to update indexes while being logged on main shop URL (you will find the error on yesterday's log).

Also FIY, I've just installed a different module from the Module Catalog and the installation went without any problem and took a few seconds. Only Faceted Search installs a few minutes and provide installation error at the end.

**Comment by khouloudbelguith** [comment]
@banan44, the whole issue is `ReceiveAckHdr: timeout 300 is exceeded`


In your php.ini file, Try to increase:

```
max_execution_time = 30 (by default is 30, try to increase it)
max_input_time = you need to increase it
memory_limit
```
Thanks to check & feedback

**Comment by banan44** [comment]
Changed these to following values:

php_value max_execution_time 500
php_value max_input_time 500
php_value memory_limit 800M

Sever informatio values changed

![obraz](https://user-images.githubusercontent.com/48493952/66997916-9c25cf00-f0d3-11e9-9a65-e5896099ab26.png)

Still "Price indexing failed" error.  What is weird the error in the log is still " ReceiveAckHdr: timeout 300 is exceeded"

**Comment by khouloudbelguith** [comment]
@banan44, could you please try to install this module, version 3.3.0 cloned from GitHub
[ps_facetedsearch.zip](https://github.com/PrestaShop/PrestaShop/files/3738712/ps_facetedsearch.zip)
Thanks to check & feedback.

**Comment by banan44** [comment]
Ok, I've tried that. No luck. After a few minutes I had following error:

![obraz](https://user-images.githubusercontent.com/48493952/67006983-ad2c0b80-f0e6-11e9-8ffd-a01e40c523cc.png)

When I refreshed page the emodule was on the list of installed modules. I've created a new template and selected Index all missing prices. Still failed.

![obraz](https://user-images.githubusercontent.com/48493952/67007508-dd27de80-f0e7-11e9-9812-145d4c40af5e.png)

I do not know what is going on. I've installed the module from your link on my other Prestashop 1.7.6.0 located on the same server but in different folder. It took about 2 minutes but the module was installed with success and the indexing works with no problem. But this is a test version with no content at all. I do not know what might block Faceted Search on my real shop.

**Comment by khouloudbelguith** [comment]
@banan44, no it is not well installed on your server.
There is something wrong during the install.
The whole issue is about the install action.
Is there any exception displayed during the install?
You have some issues on your server
![image](https://user-images.githubusercontent.com/16067358/67008757-8a035b00-f0ea-11e9-8781-5943369d49fa.png)

Thanks to check with your host & feedback.

**Comment by banan44** [comment]
@khouloudbelguith today I have tried to install the module again. The error in logs I had during installation is "content-length(6785587): ReceiveAckHdr: nothing to read from backend (LVE ID 1049), check http://docs.cloudlinux.com/mod_lsapi_troubleshooting.html". There was no red rectangle with the error like this one

![obraz](https://user-images.githubusercontent.com/48493952/67087983-5258e980-f1a4-11e9-8b03-dc34426bf66d.png)

PS just shows Service unavaible error and that's it. No other comment except the error in error log. 

I have spoken with my server administrator today. I have discussed the problem and told him that I installed the module without problem on my test Prestashop installment. He wrote:

"This message from logs says that the page scripts did not return any content when processing the request. In this situation, it would be worth reviewing some page logs if they are generated by it.
Unfortunately, we are unable to check anything more here, because the site generates nothing but this information. Did you check whether the installation process lasts 300 seconds and only then shows the error? Perhaps the process requires a longer time. As you wrote, on another page the installation worked correctly, so it's not a matter of setting up the server."

The installation process takes more that 300 seconds. Today's installation took 5 minutes until the error appeared. Administrator told me that 300 seconds is the maximum time the can set on server. I do not know where to look for different logs to verify the problem. I've also installed a bunch of different modules and all were installed without problem. Only Faceted Search is the problem. Do you have any advice or older version of this module so I could try to install it and update? 

I have uploaded today's log on my google drive https://drive.google.com/file/d/1R0qxv6HN5S2gFzwiGpJ9G_BLqJKnGKpr/view

**Comment by khouloudbelguith** [comment]
@banan44, Could you please provide me the access to your shop to test it.
This is a public space.
You can provide me the credentials by email.
My address mail: khouloud.belguith@prestashop.com
Let's be safe!

Thanks!

**Comment by banan44** [comment]
@khouloudbelguith, I've just sent you the credentials.

**Comment by khouloudbelguith** [comment]
@banan44, as always you have this error:
`Failed to load resource: the server responded with a status of 503 ()`
Could you please try to enable profiling mode
In your Projetc_Folder/config/defines.inc.php, you need to replace this

```
if (!defined('_PS_DEBUG_PROFILING_')) {
    define('_PS_DEBUG_PROFILING_', flase);
}
```

by this

```
if (!defined('_PS_DEBUG_PROFILING_')) {
    define('_PS_DEBUG_PROFILING_', true);
}
```
When profiling mode is enabled, you should have information about the time needed to run queries.

Thanks!

**Comment by banan44** [comment]
@khouloudbelguith done. What should I do now? Or you will try again?

**Comment by khouloudbelguith** [comment]
@banan44, I have this exception in your BO => Module Manager page
 
[error.pdf](https://github.com/PrestaShop/PrestaShop/files/3743969/error.pdf)

Did you make an upgrade from an old version, is it?

Thanks!

**Comment by banan44** [comment]
@khouloudbelguith you mean Prestashop or the module? I upgrade Prestashop everytime there is a new version since 1.7.2 version.

**Comment by khouloudbelguith** [comment]
@banan44, It looks like something went wrong with your update and don't have some of the database columns that were added for 1.7.6.1
The solution is to add the missing columns to your database schema.
You can find the full schema for 1.7.6.1 here: https://github.com/PrestaShop/PrestaShop/blob/1.7.6.1/install-dev/data/db_structure.sql
Thanks to check & feedback.

**Comment by banan44** [comment]
@khouloudbelguith ok, I will check entire database and get back to you. The was no error i 1-Click Upgrade. The issue with Faceted search started when I upgraded to 1.7.6.0 from 1.7.5.2. I had no error then as well. Always using the latest 1-click upgrade module.

**Comment by khouloudbelguith** [comment]
@banan44, thanks!
Waiting for your feedback.

**Comment by banan44** [comment]
@khouloudbelguith I've spent several hours to compare my database with the schema. All columns from the schema are also in my database. I have bunch of extra tables but based on their names I suspect they are for some modules I have installed in the past, like Skrill or Amazon payment (do not have these modules anymore but the tables are still there). 

Do you have any other suggestion? Maybe there is a file that normally is deleted during update but in my case it still exists and Prestashop still "sees" it and use it instead of new file? Or maybe 1-Click upgrade has a log somewhere to check if there was something missing?

**Comment by khouloudbelguith** [comment]
@banan44, Can you please provide us the log of your upgrade? It can be found in the log.txt file, available in `<admin folder>/autoupgrade/tmp/log.txt`.

THanks!

**Comment by banan44** [comment]
@khouloudbelguith I have uploaded the log to my Google Drive: https://drive.google.com/open?id=1dkLxUM1WN14ahDGJY2YaW0AhmIA5F3_A

Hope it will be helpful. It's only from the last update. There are no log files from previous upgrades.

PS. Can you remove error.pdf for security reasons?

**Comment by khouloudbelguith** [comment]
@banan44, thanks for your feedback.
As I can see in your log

>[2019-10-13 11:41:53] All files upgraded. Now upgrading database...
[2019-10-13 11:41:57] [CLEANING CACHE] File dev removed
[2019-10-13 11:42:01] [CLEANING CACHE] File prod removed
[2019-10-13 11:42:02] Migrating old setting file...
[2019-10-13 11:42:02] parameters file already exists!
[2019-10-13 11:42:02] Finished...
[2019-10-13 11:42:30] <div class="upgradeDbOk">[OK] SQL 1.7.6.1 SET SESSION sql_mode = ''</div>
[2019-10-13 11:42:30] <div class="upgradeDbOk">[OK] SQL 1.7.6.1 SET NAMES 'utf8'</div>
[2019-10-13 11:42:30] <div class="upgradeDbOk">[OK] PHP 1.7.6.1 : /* PHP:ps_1761_update_currencies(); */</div>
[2019-10-13 11:42:32] Database upgrade OK
[2019-10-13 11:43:23] Warning detected during upgrade.
[2019-10-13 11:43:23] Database upgraded. Now upgrading your Addons modules...
[2019-10-13 11:43:24] Addons modules files have been upgraded.
[2019-10-13 11:43:25] The database has been cleaned.
[2019-10-13 11:43:25] Upgrade process done

The process upgrade is well done.

Thanks!

**Comment by banan44** [comment]
@khouloudbelguith so what should I do now? You have seen yourself there is a problem with the module. Unfortunately, there is no log when I updated PS from 1.7.5.2 to 1.7.6.0. From that moment the problem with Faceted Search started.

**Comment by khouloudbelguith** [comment]
@banan44, Could you please follow this link: https://github.com/PrestaShop/php-ps-info 
phppsinfo file provides an equivalent to the phpinfo() function that reports PrestaShop Requirements information about the PHP/MySQL/Apache environment and offers suggestions for improvement.

1. Upload the file to the server. You should upload your file to the exact directory you want to test. Typically, this will be your httpdocs (/var/www/vhosts/example.com/httpdocs/) directory, although you can upload it to any subdirectory on your server as well. Use FTP to upload the file.
2. Visit the page in your browser. If you uploaded it to your html directory, you should now visit http://www.example.com/phppsinfo.php, replacing example.com with your own domain name.
3. Now you can view all of the information about PHP for your server for that particular directory.
![image](https://user-images.githubusercontent.com/16067358/44723217-a8199080-aacf-11e8-8c33-171ecbf3b5db.png)

Thanks to check and feedback.

**Comment by banan44** [comment]
@khouloudbelguith I have put that file to the main folder. This is what I have:

![fireshot](https://user-images.githubusercontent.com/48493952/67414388-48c7eb00-f5c3-11e9-852d-3bed642a6266.png)

**Comment by khouloudbelguith** [comment]
@banan44, we just released a new version of the ps_facetedearch module v3.4.0.
https://github.com/PrestaShop/ps_facetedsearch/releases/tag/v3.4.0
Could you please try to install this new version, check & feedback.

**Comment by banan44** [comment]
@khouloudbelguith still the same problem. Please take a look at the video I have made: https://drive.google.com/file/d/1P3RTfkzV3XaQI1bamn0KT1BRVqa2zbUn/view

After refresh the module is on the installed modules list but still I cannot update Indexes.

**Comment by khouloudbelguith** [comment]
@banan44, it is not well installed so, you have issues during the update of indexes.
Have you a lot of data in your shop? Suppliers, features, attributes, brands, categories?
The install is too slow in your case.

**Comment by banan44** [comment]
@khouloudbelguith yes, I have a lot of data and still growing. The database weighs over 1 GB now. 

Suppliers: 0
Brands: 34
Attributes: 0
Categories: over 500
Features: over 600
Articles: almost 1000

What should I do to install the module faster?

**Comment by khouloudbelguith** [comment]
Hi @banan44,

I tried to create a huge shop with a lot of categories ( <800) & lot of features & a lot of products (<800) & I tried to install the ps_facetedsearch module v4.4.0 & it is well installed, it takes time because we have a lot of data but it is ok.
In your case, your server unable to service your request, so it is a problem of your server.
https://drive.google.com/file/d/1VkQwN3D-KnH8Bj75cTNtSom0nMUW6dnn/view 
Despite our several trials, we could not reproduce your issue with the provided information.
It seems that your issue is not a PrestaShop's core bug but most likely a server configuration or customization problem.
I invite you to contact our support team: http://addons.prestashop.com/en/388-support to help you solve your issue.

Thanks!
